gobblin-core/src/main/java/gobblin/source/workunit/MultiWorkUnitWeightedQueue.java

```diff
@@ -111,6 +111,12 @@ public int compareTo(WeightedMultiWorkUnit weightedMultiWorkUnit) {
       return Longs.compare(this.weight, weightedMultiWorkUnit.getWeight());
     }
 
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Add `if (!(obj instanceof WeightedMultiWorkUnit) {return false;}` before doing the casting.